File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        // 2부터 n까지의 수 
        for(int i=2; i<=n; i++){ 
            
            boolean check = true;
            
            // 만약 5라고 가정 -> J:2,3까지 ! 
            // 6이라고 가정 -> 2
            for(int j=2; j<=Math.sqrt(i); j++){
                //5를 2,3으로 나누면 나누어 떨어지지 않자나 --> 소수 
                //6은 2로 나누어 떨어진다 -> 소수가 아니다 ! break
                if(i%j==0) {
                    check = false;
                    break;
                }
            }
            if(check){
                answer++;
            }
        }     
        return answer;
    }
}

File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
def solution(numbers):
    answer = 0
    num = []
    for i in range(1, len(numbers)+1):
        num.append(list(set(map(''.join, permutations(numbers, i)))))
    
    per = list(set(map(int, set(sum(num, [])))))
                   
    for p in per:
        if is_prime_number(p) == True:
                answer+=1
                  
    return answer

def is_prime_number(n):
    if n < 2:
        return False
    for i in range(2, n):
        if n%i ==0:
            return False
    return True